src/goto-instrument/code_contracts.cpp

```diff
@@ -112,17 +112,20 @@ static void check_apply_invariants(
 
   // assert the invariant
   {
-    goto_programt::targett a = havoc_code.add(
-      goto_programt::make_assertion(invariant, loop_head->source_location));
-    a->source_location.set_comment("Check loop invariant before entry");
+    auto assertion = code_assertt(invariant);
```

Comment:
Nit picking: is this use of `auto` adding value? Wouldn't `code_assertt assertion{invariant};` be even fewer characters and thus less cognitive load?